resolver/resolver.cc

```diff
@@ -1388,18 +1391,96 @@ class ResolveSignaturesWalk {
                 Exception::raise("Wrong arg count");
             }
 
+            auto *memberType = core::cast_type<core::LambdaParam>(data->resultType.get());
+            ENFORCE(memberType != nullptr);
+
+            auto lowerBound = core::Types::bottom();
+            auto upperBound = core::Types::bottom();
+
+            // NOTE: the resultType is set back in the namer to be a LambdaParam
```

Comment:
Can we set it to top & bottom in namer instead to make it not be a silent failure?

Reply:
That's how it worked before we talked last week. The issue is that it changes existing behavior in cases like `//test/testdata/todo/fixed_ordering.rb`; the error message changes, and the inferred type turns into `T::Array[<top>]` instead of `T::Array[T.untyped]`.